manticoresearch_text_embeddings.h

```diff
@@ -9,68 +9,48 @@
 #include <ostream>
 #include <new>
 
-using TextModelWrapper = void*;
+struct TextModelResult {
+  void *model;
+  char *error;
+};
 
-using LoadModelFn = TextModelWrapper(*)(const char*,
-                                        uintptr_t,
-                                        const char*,
-                                        uintptr_t,
-                                        const char*,
-                                        uintptr_t,
-                                        bool);
+using LoadModelFn = TextModelResult(*)(const char*,
+                                       uintptr_t,
+                                       const char*,
+                                       uintptr_t,
+                                       const char*,
+                                       uintptr_t,
+                                       bool);
 
-using DeleteModelFn = void(*)(TextModelWrapper);
+using FreeModelResultFn = void(*)(TextModelResult);
 
 struct FloatVec {
   const float *ptr;
   uintptr_t len;
   uintptr_t cap;
+};
 
-  bool operator==(const FloatVec& other) const {
-    return ptr == other.ptr &&
-           len == other.len &&
-           cap == other.cap;
-  }
-  bool operator!=(const FloatVec& other) const {
-    return ptr != other.ptr ||
-           len != other.len ||
-           cap != other.cap;
-  }
+struct FloatVecResult {
+  FloatVec vector;
+  char *error;
 };
 
-using MakeVectEmbeddingsFn = FloatVec(*)(const TextModelWrapper*, const char*, uintptr_t);
+using TextModelWrapper = void*;
+
+using MakeVectEmbeddingsFn = FloatVecResult(*)(const TextModelWrapper*, const char*, uintptr_t);
 
-using DeleteVecFn = void(*)(FloatVec);
+using FreeVecResultFn = void(*)(FloatVecResult);
 
 using GetLenFn = uintptr_t(*)(const TextModelWrapper*);
 
 struct EmbedLib {
   uintptr_t size;
```

Comment:
here expected version tag.
size might be good when you constantly add new functions,
but actually that is version, or unique tag of the interface, that should be changed each time something published.